internal/configload/configload.go

```diff
@@ -0,0 +1,285 @@
+		return nil, err
+	}
+
+	mongoDB, err := c.Results.MongoDB.convert()
+	if err != nil {
+		return nil, err
+	}
+
+	return &ic.Config{
+		Runner: c.Runner,
+		Dir:    c.Dir,
+		Args:   c.Args,
+		Results: ic.Results{
+			Common:   common,
+			FerretDB: ferretDB,
+			MongoDB:  mongoDB,
+		},
+	}, nil
+}
+
+// convert converts testConfig to the internal *config.TestConfig with validation.
+func (tc *testConfig) convert() (*ic.TestConfig, error) {
+	if tc == nil {
+		return nil, nil
+	}
+
+	t := ic.TestConfig{
+		Default: *tc.Default,
+		Stats:   tc.Stats.convertStats(),
+		Pass:    ic.Tests{},
+		Fail:    ic.Tests{},
+		Skip:    ic.Tests{},
+		Ignore:  ic.Tests{},
+	}
+
+	//nolint:govet // we don't care about alignment there
+	for _, testCategory := range []struct { // testCategory examples: pass, skip sections in the yaml file
+		yamlTests []any     // taken from the file, yaml representation of tests, incoming tests
+		outTests  *ic.Tests // yamlTests transformed to the internal representation
+	}{
+		{tc.Pass, &t.Pass},
+		{tc.Fail, &t.Fail},
+		{tc.Skip, &t.Skip},
+		{tc.Ignore, &t.Ignore},
+	} {
+		for _, test := range testCategory.yamlTests {
+			switch test := test.(type) {
+			case map[string]any:
+				keys := maps.Keys(test)
+				if len(keys) != 1 {
+					return nil, fmt.Errorf("invalid syntax: expected 1 element, got: %v", keys)
+				}
+
+				var arrPointer *[]string
+
+				k := keys[0]
+				switch k {
+				case "regex":
+					arrPointer = &testCategory.outTests.NameRegexPattern
+				case "not_regex":
+					arrPointer = &testCategory.outTests.NameNotRegexPattern
+				case "output_regex":
+					arrPointer = &testCategory.outTests.OutputRegexPattern
+				default:
+					return nil, fmt.Errorf("invalid field name %q", k)
+				}
+
+				mValue := test[k]
+
+				regexp, ok := mValue.(string)
+				if !ok {
+					// Arrays are illegal:
+					// - regex:
+					//   - foo
+					//   - bar
+					if _, ok := mValue.([]string); ok {
+						return nil, fmt.Errorf("invalid syntax: %s value shouldn't be an array", k)
+					}
+
+					return nil, fmt.Errorf("invalid syntax: expected string, got: %T", mValue)
+				}
+
+				// i.e. pointer to testCategory.outTests.RegexPattern = append(testCategory.outTests.RegexPattern, regexp)
+				*arrPointer = append(*arrPointer, regexp)
+
+				continue
+
+			case string:
+				testCategory.outTests.Names = append(testCategory.outTests.Names, test)
+				continue
+
+			default:
+				return nil, fmt.Errorf("invalid type of %[1]q: %[1]T", test)
+			}
+		}
+	}
+
+	return &t, nil
+}
+
+// fillAndValidate populates the configuration with default values and performs validation.
+func (c *config) fillAndValidate() error {
+	if c.Results.Common == nil {
+		c.Results.Common = &testConfig{}
+	}
+
+	validStatus := func(status ic.Status) bool {
+		s := ic.Status(strings.ToLower(string(status)))
+		_, ok := ic.KnownStatuses[s]
+
+		return ok
+	}
+
+	if c.Results.Common.Default == nil {
+		c.Results.Common.Default = new(ic.Status)
```

Comment:
`ic.Status` is not a composite type so I cannot allocate it using that syntax, right?

Reply:
@AlekSi is there a directive I can use to ignore this in the linter?